### PR TITLE
Inductive principles on set cardinality

### PR DESCRIPTION
**Pull request by pPomCo**
##### Motivation for this change

Add inductive principles based on (finite) set cardinality, e.g.
```
  (forall A, (forall B, #|B| < #|A| -> P B) -> P A)    
-----------------------------------------------------           
                   forall A, P A                                 
```
It requires Coq.Program.Wf.

##### Things done/to do

- [x] added corresponding entries in `CHANGELOG_UNRELEASED.md`
- [x] added corresponding documentation in the headers
- [x] tried to abide by the [contribution guide](https://github.com/math-comp/math-comp/blob/master/CONTRIBUTING.md)

##### Compatibility with MathComp 1.X

- [ ] ~I added the label `TODO: MC-1 port` to make sure someone ports this PR to
      the `mathcomp-1` branch **or** I already opened an issue or PR (please cross reference).~

##### Automatic note to reviewers

Read [this Checklist](https://github.com/math-comp/math-comp/wiki/Checklist-for-creating-and-review-PRs) and put a milestone if possible.

### CODE REVIEW

**Comment by gares** [comment]
Thanks for submitting a PR!


I did not look at all the lemmas you propose, but after reading the header of your file I think you should have a look at the existing "kit" for this kind of inductions. It is documented here:
https://github.com/math-comp/math-comp/blob/5ed1dbd3c891bb6dd01a8f37dd575459b3243f67/mathcomp/ssreflect/ssrnat.v#L428-L471

**Comment by CohenCyril** [comment]
@pPomCo does  @gares comment answer your problem? If so can we close this PR? Otherwise could you rephrase your need so that we understand how to address it?
Best
CC @erikmd @proux01

**Comment by erikmd** [comment]
Good point, thanks @CohenCyril !
(I had proposed Pierre P.C. to finalize this PR from my side, but I finally had lost track of this task, sorry for the delay... I'll look after it this week then)

**Comment by erikmd** [comment]
Sorry I'm confused, I straightforward replied to the notification believing you were talking about PR https://github.com/math-comp/math-comp/pull/931
(so you may ignore my previous comment that wasn't related to this PR #1120)

**Comment by affeldt-aist** [comment]
I propose that we move this PR to the next milestone because it is not clear that the proposed lemmas fit the desired pattern, there is no example using the lemmas to further discuss their usefulness, and they are anyway in an independent file so they will continue to compile for a while.
